### PR TITLE
ce trace ouput in function mode

### PR DESCRIPTION
**Pull request by ChenfengWei0**
issue: #2083 

```c
#include <assert.h>

void foo(unsigned int x)
{
	x++;
	if ((10 * x - 1) == 1001)
		assert(0);
}

int main()
{
	foo(3435973937);
}
```

Command: `esbmc --function foo`

Output
```
[Counterexample]


State 1 file 1.c line 5 column 3 function foo thread 0
----------------------------------------------------
  x = 3435973936 (11001100 11001100 11001101 00110000)

State 2 file 1.c line 5 column 3 function foo thread 0
----------------------------------------------------
  x = 3435973937 (11001100 11001100 11001101 00110001)

State 3 file 1.c line 7 column 5 function foo thread 0
----------------------------------------------------
Violated property:
  file 1.c line 7 column 5 function foo
  assertion 0
  0


VERIFICATION FAILED
```

### CODE REVIEW

**Review by mikhailramalho — APPROVED** [review]
Interesting approach. LGTM.

Nit: the location of the assignment is not correct (it should be line 3), I'd say either try to get the location from the argument symbol or don't use any location at all.

**Comment by ChenfengWei0** [comment]
> Interesting approach. LGTM.
> 
> Nit: the location of the assignment is not correct (it should be line 3), I'd say either try to get the location from the argument symbol or don't use any location at all.

Changed as suggested.
```
State 1  thread 0
----------------------------------------------------
  x = 3435973936 (11001100 11001100 11001101 00110000)

State 2 file 1.c line 4 column 2 function foo thread 0
----------------------------------------------------
  x = 3435973937 (11001100 11001100 11001101 00110001)

State 3 file 1.c line 6 column 3 function foo thread 0
----------------------------------------------------
Violated property:
  file 1.c line 6 column 3 function foo
  assertion 0
  0
```

File: src/goto-programs/assign_params_as_non_det.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  goto_programt &goto_program = F.second.body;
+  auto it = (goto_program).instructions.begin();
+
+  for (const auto &arg : t.arguments())
+  {
+    // lhs
+    const auto &_id = arg.get("#identifier");
+    if (context.find_symbol(_id) == nullptr)
+      return false; // Not expected
+    exprt lhs = symbol_expr(*context.find_symbol(_id));
```
Comment (mikhailramalho) [inline src/goto-programs/assign_params_as_non_det.cpp:37]: can you check if the symbol returned by `context.find_symbol(_id)` has the location of the parameter declaration?

Comment (ChenfengWei0) [reply src/goto-programs/assign_params_as_non_det.cpp:37]: Thanks for the suggestion. The location is now
```sh
[Counterexample]


State 1 file 1.c line 3 column 1 thread 0
----------------------------------------------------
  x = 0 (00000000 00000000 00000000 00000000)

State 2 file 1.c line 5 column 2 function foo thread 0
----------------------------------------------------
  x = 1 (00000000 00000000 00000000 00000001)

State 3 file 1.c line 6 column 2 function foo thread 0
----------------------------------------------------
Violated property:
  file 1.c line 6 column 2 function foo
  assertion x > 3
  x > 3
```

**Review by lucasccordeiro — APPROVED** [review]

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

**Pull request merged**